### PR TITLE
Decoders improvements 3 - another batch

### PR DESCRIPTION
**Pull request by jrouaix**
just some more refacto, mainly on `network` & `opendirectory` modules

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -64,13 +63,36 @@ pub(crate) fn extract_string_size(data: &[u8], message_size: u64) -> nom::IResul
     Ok((input, String::from("Could not find path string")))
 }
 
+const NULL_BYTE: u8 = 0;
+
+/// Extract an UTF8 string from a byte array, stops at NULL_BYTE or END OF STRING
+/// Consumes the end byte
+pub(crate) fn cstring(input: &[u8]) -> nom::IResult<&[u8], String> {
```
Comment (puffyCid) [inline src/util.rs:70]: According to clippy, this function is unused?
https://github.com/mandiant/macos-UnifiedLogs/actions/runs/12357237427/job/34485017955
Is this expected? Seems to do same thing as `non_empty_cstring`?

File: src/decoders/network.rs [hunk 2]
```diff
@@ -5,159 +5,117 @@
...
-    map(be_u32, |val| Ipv4Addr::from(val).to_string())(input)
+pub(crate) fn get_ip_four(input: &[u8]) -> nom::IResult<&[u8], Ipv4Addr> {
+    map(be_u32, |val| Ipv4Addr::from(val))(input)
 }
 
 /// Get the IPv6 data
-pub(crate) fn get_ip_six(input: &[u8]) -> nom::IResult<&[u8], String> {
-    map(be_u128, |val| Ipv6Addr::from(val).to_string())(input)
+pub(crate) fn get_ip_six(input: &[u8]) -> nom::IResult<&[u8], Ipv6Addr> {
+    map(be_u128, |val| Ipv6Addr::from(val))(input)
```
Comment (puffyCid) [inline src/decoders/network.rs:118]: Clippy generates a few warnings in `network.rs` , could u address them?

**Review by puffyCid — COMMENTED** [review]
Looks good thanks!
Left few comments for very small clippy nitpicks

**Comment by jrouaix** [comment]
Should be good now, paired with clippy

**Pull request merged**